packages/components/src/core/Dropdown/index.tsx

```diff
@@ -238,26 +290,46 @@ const Dropdown = <
       return;
     }
 
+    if (multiple) {
+      setValue(pendingValue);
+    }
+
     if (anchorEl) {
       anchorEl.focus();
     }
 
-    if (closeOnBlur) onClose?.();
+    if (closeOnBlur) onClose?.(event, reason);
 
     if (shouldShowButtons) {
-      onClose?.();
+      onClose?.(event, reason);
       setOpen(false);
     }
   }
 
   function handleCancel() {
+    if (multiple) {
+      // (masoudmanson): To undo the latest actions made on the selections,
+      // we set the value of the selection to the pendingValue. This allows us to
+      // cancel any recent changes and restore the previous selection state.
+      setPendingValue(value);
+    }
+
     if (anchorEl) {
       anchorEl.focus();
     }
 
-    onClose?.();
```

Comment:
I don't have context, but just wanted to make sure this removal is intentional 😆

Reply:
We required two arguments for the `onClose` function, which are obtained from MUI's Autocomplete `onClose`, like this: `onClose?.(event, reason);`
However, in the `handleCancel()` function, we lacked those `event` and `reason`, making it impossible to properly emit the `onClose` event.

Reply:
Ahh I see that makes sense now thanks for taking the time to explain!